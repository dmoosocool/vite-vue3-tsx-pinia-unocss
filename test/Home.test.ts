/* eslint-disable @typescript-eslint/no-explicit-any */
import { shallowMount } from '@vue/test-utils'
import { expect, it, describe, beforeAll } from 'vitest'
import { createPinia } from 'pinia'

import Home from '@/views/home'

let wrapper: any
beforeAll(() => {
  wrapper = shallowMount(Home, {
    global: {
      plugins: [createPinia()],
    },
  })
})

describe('tsx vue pinia', () => {
  it('test mounted', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('test events and pinia', async () => {
    const plusBtn = wrapper.find('button.plus-btn')
    const minusBtn = wrapper.find('button.minus-btn')
    await plusBtn.trigger('click')
    expect(wrapper.html()).toContain('<p>your counter: 1</p>')
    await minusBtn.trigger('click')
    expect(wrapper.html()).toContain('<p>your counter: 0</p>')
  })
})
