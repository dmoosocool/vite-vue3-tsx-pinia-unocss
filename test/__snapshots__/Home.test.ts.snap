// Vitest Snapshot v1

exports[`tsx vue pinia > test mounted 1`] = `
"<div class=\\"container mx-auto\\">
  <div class=\\"hstack\\">
    <h1>Hello dmoosocool ~</h1>
    <p>your counter: 0</p>
  </div>
  <div class=\\"hstack\\" rss3:justify=\\"between\\"><button class=\\"plus-btn\\">+</button><button class=\\"minus-btn\\">-</button></div>
</div>"
`;
